dvc/utils/diff.py

```diff
@@ -26,6 +26,8 @@ def _diff_vals(old, new, with_unchanged):
     res = {"old": old, "new": new}
     if isinstance(new, (int, float)) and isinstance(old, (int, float)):
         res["diff"] = new - old
+    elif isinstance(new, str) and isinstance(old, str):
+        res["diff"] = ""
```

Comment:
I don't think this is really needed. The default behavior when `diff` entry is missing should be just fine, as a placeholder `-` gets rendered in the UI.